src/com/amazon/corretto/crypto/provider/Utils.java

```diff
@@ -59,6 +64,29 @@ private Utils() {
      */
     static native int getDigestLength(long evpMd);
 
+    static String jceDigestNameToAwsLcName(String jceName) {
+        if (jceName == null) {
+            return null;
+        }
+        // e.g. "SHA-512/256" => "SHA512-256"
+        return jceName.replace("-", "").replace("/", "-").toUpperCase();
+    }
+
+    static long getMdPtr(String digestName) {
+        final String name = jceDigestNameToAwsLcName(digestName);
+        // NOTE: it's a little awkward to throw an unchecked exception here, but the lambda
+        // invoking the native Utils.getEvpMdFromName can also throw an unchecked exception,
+        // so callers already need to handle this.
+        if (!name.startsWith("SHA")) {
+            throw new RuntimeException("Unsupported digest algorithm");
```

Comment:
This should be an `IllegalArgumentException`

Reply:
ah, yep. agreed. done.